hive.yaml

```diff
@@ -0,0 +1,102 @@
+package:
+  name: hive
+  version: 4.0.0
+  epoch: 0
+  description: Apache Hive
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - openjdk-8
+
+environment:
+  contents:
+    packages:
+      - bash
+      - busybox
+      - maven
+      - openjdk-8
+  environment:
+    LANG: en_US.UTF-8
+    JAVA_HOME: /usr/lib/jvm/java-1.8-openjdk
+    HIVE_VERSION: 4.0.0
+    HADOOP_VERSION: 3.3.6
+    TEZ_VERSION: 0.10.4
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/apache/hive
+      tag: rel/release-${{package.version}}
+      expected-commit: 183f8cb41d3dbed961ffd27999876468ff06690c # rel/release-4.0.0
+
+  - uses: patch
+    with:
+      # This patch covers license check for the goal org.apache.rat:apache-rat-plugin:0.13:check
+      # because the .gitconfig file for safe directories at build time should be ignored.
+      patches: rat.patch
+
+  - runs: |
+      mvn package -DskipTests -Dmaven.javadoc.skip=true -Pdist \
+        -Dhadoop.version="${HADOOP_VERSION}" \
+        -Dtez.version="${TEZ_VERSION}"
+      mkdir -p ${{targets.destdir}}/usr/share/java/hive
+      tar -xf packaging/target/apache-hive-${{package.version}}-bin.tar.gz \
+        --strip-component=1 -C ${{targets.destdir}}/usr/share/java/hive
+
+test:
+  environment:
+    contents:
+      packages:
+        - bash
+        - gpg
+        - gpg-agent
+        - gnupg-dirmngr
+        - curl
+        - libnss
+        - libnss-dev
+    environment:
+      JAVA_HOME: /usr/lib/jvm/java-1.8-openjdk
+      HADOOP_VERSION: 3.3.6
+      HIVE_HOME: /usr/share/java/hive
+  pipeline:
+    - name: Download Hadoop
+      runs: |
+        gpg --keyserver hkps://keyserver.ubuntu.com --recv-key CD32D773FF41C3F9E74BDB7FB362E1C021854B9D
+        curl -sLO https://dlcdn.apache.org/hadoop/common/hadoop-${HADOOP_VERSION}/hadoop-${HADOOP_VERSION}.tar.gz.asc
+        curl -sLO https://dlcdn.apache.org/hadoop/common/hadoop-${HADOOP_VERSION}/hadoop-${HADOOP_VERSION}.tar.gz
+        gpg --verify hadoop-${HADOOP_VERSION}.tar.gz.asc hadoop-${HADOOP_VERSION}.tar.gz
+        tar -zxf hadoop-${HADOOP_VERSION}.tar.gz && rm *.tar.gz*
+    - name: Prepare Hadoop standalone
+      runs: |
+        echo "JAVA_HOME=/usr/lib/jvm/java-1.8-openjdk" >> hadoop-${HADOOP_VERSION}/etc/hadoop-env.sh
+        echo "127.0.0.1 $(hostname)" >> /etc/hosts
+    # This test suite starts Beeline and HiveServer2 in the same process.
+    # Ref: https://cwiki.apache.org/confluence/display/Hive/HiveServer2+Overview
+    - name: Create a table in Hive
+      runs: |
+        export HADOOP_HOME=$(pwd)/hadoop-${HADOOP_VERSION}
+
+        # Prepare HDFS directories
+        echo "build:x:$(id -u):$(id -g):build:/home/build:/bin/sh:" >> /etc/passwd
```

Comment:
This tweak is there because Hadoop uses passwd DB to get the user name during authentication and access control checks.